apps/cnquery/cmd/login.go

```diff
@@ -105,15 +123,23 @@ func register(token string, annotations map[string]string, timer int, splay int,
 		claims, err := upstream.ExtractTokenClaims(token)
 		if err != nil {
 			log.Warn().Err(err).Msg("could not read the token")
+			return tokenValidationErr
 		} else {
 			if len(claims.Description) > 0 {
 				log.Info().Msg("token description: " + claims.Description)
 			}
 			if claims.IsExpired() {
-				log.Warn().Msg("token is expired")
+				return tokenExpiredErr
+			} else if claims.Expiry == nil {
+				log.Warn().Msg("token does not contain an expiry date")
 			} else {
 				log.Info().Msg("token will expire at " + claims.Claims.Expiry.Time().Format(time.RFC1123))
 			}
+			if claims.Space == "" {
```

Comment:
The issue also mentions checking for `scope`.

Reply:
There is no `scope` claim in the claims

Reply:
Perhaps the issue means this?
https://github.com/mondoohq/cnquery/blob/main/cli/config/config.go#L272-L287

Reply:
This is read from yml config / env variables and used if there is no token provided
Do you think we should check it like `if (config).Scope == "" && (tokenClaims).Space == "" { ... }` here?

Reply:
I double-checked the issue. It is explicitly stating the claim and also has an example:
> If it is not a token that includes either scope or space claim, then issue the following error

@chris-rock : Can you please provide steps how you produced the token?

Reply:
Current Claims definition

```go
type VerifyClaim struct { // size=192 (0xc0)
    jwt.Claims
    CustomTokenClaims
}
// Embedded fields:
Issuer         string            // through Claims            
Subject        string            // through Claims            
Audience       jwt.Audience      // through Claims            
Expiry         *jwt.NumericDate  // through Claims            
NotBefore      *jwt.NumericDate  // through Claims            
IssuedAt       *jwt.NumericDate  // through Claims            
ID             string            // through Claims            
Space          string            // through CustomTokenClaims 
Description    string            // through CustomTokenClaims 
ApiEndpoint    string            // through CustomTokenClaims 
Labels         map[string]string // through CustomTokenClaims 
Owner          string            // through CustomTokenClaims 
CertValidUntil time.Time         // through CustomTokenClaims
```

Reply:
Oh, ok. :+1: